provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLInstance.java

```diff
@@ -0,0 +1,152 @@
+   */
+  public enum GoogleSQLInstanceDisplayPropertyToken implements DisplayPropertyToken {
+
+    //TODO(kl3n1nz) Add other display properties.
+    /**
+     * The display property.
+     */
+    DATABASE_INSTANCE_ID(new SimpleDisplayPropertyBuilder()
+        .displayKey("DatabaseInstanceId")
+        .defaultDescription("The ID of the database instance.")
+        .sensitive(false)
+        .build()) {
+      @Override
+      protected String getPropertyValue(DatabaseInstance instance) {
+        return instance.getName();
+      }
+    };
+
+    private final DisplayProperty displayProperty;
+
+    /**
+     * Creates a Google instance display property token with the specified parameters.
+     *
+     * @param displayProperty the display property
+     */
+    GoogleSQLInstanceDisplayPropertyToken(DisplayProperty displayProperty) {
+      this.displayProperty = displayProperty;
+    }
+
+    /**
+     * Returns the value of the property from the specified instance.
+     *
+     * @param instance the instance
+     * @return the value of the property from the specified instance
+     */
+    protected abstract String getPropertyValue(DatabaseInstance instance);
+
+    @Override
+    public DisplayProperty unwrap() {
+      return displayProperty;
+    }
+  }
+
+  public static final Type TYPE = new ResourceType("GoogleCloudSQLInstance");
+
+  /**
+   * Creates a Google Cloud SQL instance with the specified parameters.
+   *
+   * @param template        the template from which the instance was created
+   * @param instanceId      the instance identifier
+   * @param instanceDetails the provider-specific instance details
+   * @throws IllegalArgumentException if the instance does not have a valid private IP address
+   */
+  protected GoogleCloudSQLInstance(GoogleCloudSQLInstanceTemplate template, String instanceId,
+      DatabaseInstance instanceDetails) {
+    // TODO(kl3n1nz) Might want to let the user specify a port number.
+    super(template, instanceId, getPrivateIpAddress(instanceDetails), 3306, instanceDetails);
+  }
+
+  /**
+   * Returns the private IP address of the specified Google instance.
+   *
+   * @param instance the instance
+   * @return the private IP address of the specified Google instance
+   * @throws IllegalArgumentException if the instance does not have a valid private IP address
+   */
+  private static InetAddress getPrivateIpAddress(DatabaseInstance instance) {
+    Preconditions.checkNotNull(instance, "instance is null");
+
+    List<IpMapping> ipMappingList = instance.getIpAddresses();
+
+    if (ipMappingList == null || ipMappingList.size() == 0) {
+      throw new IllegalArgumentException("No network interfaces found for database instance '" + instance.getName() + "'.");
+    } else {
+      try {
+        return InetAddress.getByName(ipMappingList.get(0).getIpAddress());
+      } catch (UnknownHostException e) {
+        throw new IllegalArgumentException("Invalid IPv4 address", e);
```

Comment:
I don't think it should be consistent in this case. GCE instance has 2 types of IP addresses (public and private) and it was emphasized. For Google Cloud SQL there is only one type of IPv4 address.